src/main/java/ch/njol/skript/expressions/ExprNamed.java

```diff
@@ -35,51 +35,85 @@
-		"If you want to (re)name existing items/inventories you can either use this expression or use <code>set <a href='#ExprName'>name of &lt;item/inventory&gt;</a> to &lt;text&gt;</code>.")
-@Examples({"give a diamond sword of sharpness 100 named \"&lt;gold&gt;Excalibur\" to the player",
-		"set tool of player to the player's tool named \"&lt;gold&gt;Wand\"",
-		"set the name of the player's tool to \"&lt;gold&gt;Wand\"",
-		"open hopper inventory named \"Magic Hopper\" to player"})
+@Description({
+	"Directly names an item/inventory, useful for defining a named item/inventory in a script.",
+	"If you want to (re)name existing items/inventories you can either use this expression " +
+			"or use <code>set <a href='#ExprName'>name of &lt;item/inventory&gt;</a> to &lt;text&gt;</code>."
+})
+@Examples({
+	"give a diamond sword of sharpness 100 named \"&lt;gold&gt;Excalibur\" to the player",
+	"set tool of player to the player's tool named \"&lt;gold&gt;Wand\"",
+	"set the name of the player's tool to \"&lt;gold&gt;Wand\"",
+	"open hopper inventory named \"Magic Hopper\" to player"
+})
 @Since("2.0, 2.2-dev34 (inventories)")
 public class ExprNamed extends PropertyExpression<Object, Object> {
+
 	static {
-		Skript.registerExpression(ExprNamed.class, Object.class, ExpressionType.PROPERTY,
+		Skript.registerExpression(ExprNamed.class, Object.class, ExpressionType.COMBINED,
 				"%itemtype/inventorytype% (named|with name[s]) %string%");
 	}
-	
-	@SuppressWarnings("null")
+
 	private Expression<String> name;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr(exprs[0]);
 		name = (Expression<String>) exprs[1];
+		if (getExpr() instanceof Literal) {
+			Literal<?> literal = (Literal<?>) getExpr();
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't create a '" + literal.toString() + "' inventory. It's not creatable!");
+				return false;
+			}
+		}
 		return true;
 	}
-	
+
 	@Override
-	protected Object[] get(final Event e, final Object[] source) {
-		String name = this.name.getSingle(e);
+	protected Object[] get(Event event, Object[] source) {
+		String name = this.name.getSingle(event);
 		if (name == null)
-			return get(source, obj -> obj); // No name provided, do nothing
+			return get(source, object -> {
+				if (object instanceof InventoryType) {
+					try {
+						return Bukkit.createInventory(null, (InventoryType) object);
+					} catch (NullPointerException | IllegalArgumentException e) {
+						// Spigot forgot to label some InventoryType's as non creatable in some versions < 1.19.4
```

Comment:
Since this was fixed in a later versions and we know it is the CHISELED_BOOKSHELF or COMPOSTER, I think we should switch these cases rather than running a try/catch.

Reply:
I don't understand what you mean. Are you meaning directly comparing against the InventoryType enum? Is there an issue with using a try catch?